week7/answers.sql

```diff
@@ -1,3 +1,314 @@
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE items (
+	item_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	name VARCHAR(30),
+	armor INT UNSIGNED NOT NULL,
+	damage INT UNSIGNED NOT NULL
+);
+
+CREATE TABLE inventory (
+	inventory_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT inventory_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT inventory_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE equipped (
+	equipped_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT equipped_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT equipped_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+DELIMITER ;;
+CREATE PROCEDURE equip(inventory_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT inventory_id INTO find_item_by_id;
+
+	SELECT 
+		inventory.character_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		inventory.item_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO equipped (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM inventory WHERE inventory.inventory_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE unequip(equipped_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT equipped_id INTO find_item_by_id;
+
+	SELECT 
+		equipped.character_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		equipped.item_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO inventory (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM equipped WHERE equipped.equipped_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE set_winners(team_id INT UNSIGNED)
+BEGIN
+	DECLARE find_winner_by_team_id INT UNSIGNED;
+	SELECT team_id INTO find_winner_by_team_id;
+
+	DELETE FROM winners;
+
+	INSERT INTO winners
+	SELECT 
+		characters.character_id,
+		characters.name
+	FROM teams
+		INNER JOIN team_members
+			ON teams.team_id = team_members.team_id
+		INNER JOIN characters
+			ON team_members.character_id = characters.character_id
+	WHERE teams.team_id = find_winner_by_team_id;
+END;;
+
+CREATE FUNCTION armor_total(character_id INT UNSIGNED)
+RETURNS TINYINT
```

Comment:
Noticed that you use a INT to represent the item armor while returning a `TINYINT` here. While the item armor implies a value could be more than 2 billion (this is fine, by the way), the signature of this function implies that, after all armor is added up, that a character could never have more than 127 total armor. This works because the numbers are small, but from a new-to-the-project developer perspective, they might wonder how:

`{big number (INT)} + {big number (INT)} + {big number (INT)} = {small number (TINYINT)}`

I would recommend either changing all of the armor declarations to `TINYINT` or changing the function return to an `INT` for consistency.